src/pynwb/core.py

```diff
@@ -325,82 +114,6 @@ def __init__(self, **kwargs):
-
-    @docval({'name': 'name', 'type': str, 'doc': 'the name of this VectorData'},
-            {'name': 'data', 'type': ('array_data', 'data'),
-             'doc': 'a dataset where the first dimension is a concatenation of multiple vectors'},
-            {'name': 'target', 'type': NWBData, 'doc': 'the target dataset that this index applies to'})
-    def __init__(self, **kwargs):
-        call_docval_func(super(Index, self).__init__, kwargs)
-
-
-@register_class('VectorData', CORE_NAMESPACE)
-class VectorData(NWBData):
-
-    __nwbfields__ = ("description",)
-
-    @docval({'name': 'name', 'type': str, 'doc': 'the name of this VectorData'},
-            {'name': 'description', 'type': str, 'doc': 'a description for this column'},
-            {'name': 'data', 'type': ('array_data', 'data'),
-             'doc': 'a dataset where the first dimension is a concatenation of multiple vectors', 'default': list()})
-    def __init__(self, **kwargs):
-        call_docval_func(super(VectorData, self).__init__, kwargs)
-        self.description = getargs('description', kwargs)
-
-    @docval({'name': 'val', 'type': None, 'doc': 'the value to add to this column'})
-    def add_row(self, **kwargs):
-        val = getargs('val', kwargs)
-        self.data.append(val)
-
-
-@register_class('VectorIndex', CORE_NAMESPACE)
-class VectorIndex(Index):
-
-    @docval({'name': 'name', 'type': str, 'doc': 'the name of this VectorIndex'},
-            {'name': 'data', 'type': ('array_data', 'data'),
-             'doc': 'a 1D dataset containing indexes that apply to VectorData object'},
-            {'name': 'target', 'type': VectorData, 'doc': 'the target dataset that this index applies to'})
-    def __init__(self, **kwargs):
-        call_docval_func(super(VectorIndex, self).__init__, kwargs)
-        self.target = getargs('target', kwargs)
-
-    def add_vector(self, arg):
-        self.target.extend(arg)
-        self.data.append(len(self.target))
-
-    def add_row(self, arg):
-        self.add_vector(arg)
-
-    def __getitem_helper(self, arg):
-        start = 0 if arg == 0 else self.data[arg-1]
-        end = self.data[arg]
-        return self.target[start:end]
-
-    def __getitem__(self, arg):
-        if isinstance(arg, slice):
-            indices = list(range(*arg.indices(len(self.data))))
-            ret = list()
-            for i in indices:
-                ret.append(self.__getitem_helper(i))
-            return ret
-        else:
-            return self.__getitem_helper(arg)
-
-
-@register_class('ElementIdentifiers', CORE_NAMESPACE)
-class ElementIdentifiers(NWBData):
-
-    @docval({'name': 'name', 'type': str, 'doc': 'the name of this ElementIdentifiers'},
-            {'name': 'data', 'type': ('array_data', 'data'), 'doc': 'a 1D dataset containing identifiers',
-             'default': list()})
-    def __init__(self, **kwargs):
-        call_docval_func(super(ElementIdentifiers, self).__init__, kwargs)
-
-
 class NWBTable(NWBData):
```

Comment:
What do NWBTable and NWBTableRegion do? They are not in the core schema. They are not used by DynamicTable or DynamicTableRegion. From the comments, it looks like the ElectrodeTable used to be an NWBTable. Are these classes effectively dead code? If so, can we remove or deprecate their usage in favor of DynamicTable and DynamicTableRegion?

Reply:
They are not they’re own type. NWBTable is a class for handling 1D datasets with a compound type, which has table like behavior

Reply:
Since it seems pretty generic (and not in the core NWB schema), would it make sense to move this into HDMF common? That could be a separate PR.